src/Tasks.UnitTests/CodeTaskFactoryTests.cs

```diff
@@ -1120,6 +1124,265 @@ public void RedundantMSBuildReferences()
             MockLogger mockLogger = Helpers.BuildProjectWithNewOMExpectSuccess(projectFileContents);
             mockLogger.AssertLogContains("Hello, World!");
         }
+
+        [Fact]
+        public void EmbedsGeneratedFromSourceFileInBinlog()
```

Comment:
A lot of code seems to be duplicated in these tests. Any chance the common logic could be moved into a shared parameterized helper?

Reply:
Done